core/src/mast/mod.rs

```diff
@@ -60,6 +62,63 @@ impl MastForest {
         Ok(new_node_id)
     }
 
+    /// Adds a basic block node to the forest, and returns the [`MastNodeId`] associated with it.
+    pub fn add_block(
+        &mut self,
+        operations: Vec<Operation>,
+        decorators: Option<DecoratorList>,
+    ) -> Result<MastNodeId, MastForestError> {
+        match decorators {
+            Some(decorators) => {
+                self.add_node(MastNode::new_basic_block_with_decorators(operations, decorators))
+            }
+            None => self.add_node(MastNode::new_basic_block(operations)),
+        }
+    }
+
+    /// Adds a join node to the forest, and returns the [`MastNodeId`] associated with it.
+    pub fn add_join(
+        &mut self,
+        left_child: MastNodeId,
+        right_child: MastNodeId,
+    ) -> Result<MastNodeId, MastForestError> {
+        self.add_node(MastNode::new_join(left_child, right_child, self))
+    }
```

Comment:
One potentially nice thing to validate here would be that `left_child` and `right_child` have smaller values than the newly created node ID. We could create a new error variant in `MastForestError` for this.

This is also relevant for `add_split()`, `add_loop()`, `add_call()`, and `add_syscall()` methods below.

Reply:
Updated.

LMK if you would like the error string to be more descriptive and contain references to the type of node and other IDs.